snuba/migrations/snuba_migrations/transactions/0015_transactions_add_source_column.py

```diff
@@ -0,0 +1,52 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, String
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    The source column is required to query for transaction events
+    in the transaction summary when the transaction name has been
+    redacted in metrics.
+    """
+
+    blocking = False
+
+    def __forward_migrations(
+        self, table_name: str
+    ) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name=table_name,
+                column=Column(
+                    "transaction_source",
+                    String(Modifiers(nullable=True, low_cardinality=True)),
```

Comment:
The linked processor PR is defaulting to an empty string. So I don't think this needs to be nullable. Using nullable takes more storage space and thus affects performance.

Reply:
fixed.